### PR TITLE
feat: fullstack solution

### PR DESCRIPTION
**Pull request by surajsbharadwaj**
### Description

1. Downgrade terraform ibm provider version to 1.49.0 to make it compatible with secure landing zone vpc module.
2. Reorganized the example folder structure
3. Upgraded the examples to have a full stack solution (secure landing zone vpc plus Powervs infra deployed in one go).
4. Renamed the flavour for IBM catalog

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [x] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

1. Downgrade terraform ibm provider version to 1.49.0 to make it compatible with secure landing zone vpc module.
2. Reorganized the example folder structure
3. Upgraded the examples to have a full stack solution (secure landing zone vpc plus Powervs infra deployed in one go).
4. Renamed the flavour for IBM catalog

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Comment by surajsbharadwaj** [comment]
@kbiegert could you please review the ibm_catalog.json file and the descriptions in it ?
@Beilmann  and @SirSpidey  **Docs:**
could you guys review :
- readme files especially the descriptions 
- Description in the variables file (under examples folder (examples/ibm-catalog/deployable-architectures/full-stack/variables.tf)

File: examples/ibm-catalog/deployable-architectures/full-stack/README.md [hunk 1]
```diff
@@ -1,73 +1,71 @@
 # IBM Cloud catalog example for Power infrastructure for regulated industries
 
-This example illustrates how to use the `power-infrastructure` module in the IBM catalog. It provisions the following infrastructure:
+This example provisions the following infrastructure:
+- A VPC Infrastructure with following components
+    -  3 VPCs with 1 VSI in each VPC( 1 Management(jump/bastion) vsi, 1 inet-svs vsi configured as squid proxy server, 1 private-svs vsi configured as NFS, NTP, DNS server)
+    - Installs and configures the Squid Proxy, DNS Forwarder, NTP Forwarder and NFS on hosts, and sets the host as server for these services NTP, NFS, DNS using ansible roles.
```
Comment (SirSpidey) [inline examples/ibm-catalog/deployable-architectures/full-stack/README.md:6]:
```suggestion
This example sets up the following infrastructure:
- A VPC Infrastructure with the following components:
    -  Provisions three VPCs with one VSI in each VPC (one management(jump/bastion) VSI, one inet-svs VSI configured as squid proxy server, one private-svs VSI configured as NFS, NTP, DNS server).
    - Installs and configures the Squid Proxy, DNS Forwarder, NTP forwarder and NFS on hosts, and sets the host as the server for the NTP, NFS, and DNS services by using Ansible roles.
```

Made the list items parallel by starting with a verb, although I'm not wed to those verbs.

File: examples/ibm-catalog/deployable-architectures/full-stack/README.md [hunk 2]
```diff
@@ -1,73 +1,71 @@
...
-    - Creates one or two IBM Cloud connections with an option to reuse the connections
-    - Attaches the IBM Cloud connections to a transit gateway
-    - Attaches the private networks to the IBM Cloud connections
-    - Creates an SSH key
-    - Installs and configures the Squid Proxy, DNS Forwarder, NTP Forwarder and NFS on specified host, and sets the host as server for these services
+    - Creates two private networks: a management network and a backup network.
+    - Creates one or two IBM Cloud connections with an option to reuse the connections.
+    - Attaches the IBM Cloud connections to a transit gateway.
+    - Attaches the private networks to the IBM Cloud connections.
+    - Creates a SSH key.
```
Comment (SirSpidey) [inline examples/ibm-catalog/deployable-architectures/full-stack/README.md:13]:
```suggestion
    - Creates an SSH key.
```

File: examples/ibm-catalog/deployable-architectures/full-stack/README.md [hunk 3]
```diff
@@ -1,73 +1,71 @@
...
 
-## Before you begin
+## Reference architectures
```
Comment (SirSpidey) [inline examples/ibm-catalog/deployable-architectures/full-stack/README.md:15]: I think if you want to link to the diagram only, we should rename the section to **Architecture diagram**. To leave the heading, link to the `.md` file.

**Review by SirSpidey — CHANGES_REQUESTED** [review]
couple of small suggestions

**Comment by kbiegert** [comment]
I was thinking that the plan was going to be to retain the power infrastructure as a add on (keep it as is) and implement a new terraform that would represent the full stack by invoking the power infra and SLZ.  Is there a new plan?

**Comment by surajsbharadwaj** [comment]
@kbiegert . In this release let's do it for full stack. On next pull request i will add the modified and more sensible code for extension.

**Comment by Beilmann** [comment]
@surajsbharadwaj - you changed SLZ extension to full stack. We plan to have 2 variants: SLZ extension + full stack. Why not to keep also the existing one?

**Comment by surajsbharadwaj** [comment]
@Beilmann let's discuss on Monday. We need 2 architecture diagrams. Variable description altered and may be more in detail about prerequisites. For this release it is full stack solution only. I think will have to recolor the architecture diagram to represent the full stack solution

**Comment by Beilmann** [comment]
@surajsbharadwaj - names sind ok, from my point of view. Will rename and extend reference architectures in the same way.

**Comment by surajsbharadwaj** [comment]
invalid. A new one will be created

**Pull request closed without merging**